core/examples/wasm.rs

```diff
@@ -1,60 +1,25 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
-use deno_core::op;
+use deno_core::include_ascii_string;
+use deno_core::op2;
 use deno_core::Extension;
 use deno_core::JsRuntime;
 use deno_core::Op;
+use deno_core::OpState;
 use deno_core::RuntimeOptions;
-use std::mem::transmute;
-use std::ptr::NonNull;
+use deno_core::WasmMemory;
 
-// This is a hack to make the `#[op]` macro work with
-// deno_core examples.
-// You can remove this:
-
-use deno_core::*;
-
-struct WasmMemory(NonNull<v8::WasmMemoryObject>);
-
-fn wasm_memory_unchecked(state: &mut OpState) -> &mut [u8] {
-  let WasmMemory(global) = state.borrow::<WasmMemory>();
-  // SAFETY: `v8::Local` is always non-null pointer; the `HandleScope` is
-  // already on the stack, but we don't have access to it.
-  let memory_object = unsafe {
-    transmute::<NonNull<v8::WasmMemoryObject>, v8::Local<v8::WasmMemoryObject>>(
-      *global,
-    )
-  };
-  let backing_store = memory_object.buffer().get_backing_store();
-  let ptr = backing_store.data().unwrap().as_ptr() as *mut u8;
-  let len = backing_store.byte_length();
-  // SAFETY: `ptr` is a valid pointer to `len` bytes.
-  unsafe { std::slice::from_raw_parts_mut(ptr, len) }
-}
-
-#[op(wasm)]
-fn op_wasm(state: &mut OpState, memory: Option<&mut [u8]>) {
-  let memory = memory.unwrap_or_else(|| wasm_memory_unchecked(state));
+#[op2(fast)]
+fn op_wasm(state: &mut OpState, #[smi] rid: u32, #[memory] memory: WasmMemory) {
```

Comment:
I think this particular construct is a bit too awkward -- what if we make `#[memory]` a non-virtual argument that takes the handle?

In fast mode, the handle will redundant and the code can ignore it. In slow mode, it can automatically get the wasm memory from the state and then give you a buffer from it.

It would look something like this:

```
#[op2(fast)]
fn op_wasm(#[memory] memory: &[u8]) {
 ...
}
```

Reply:
It wouldn't be a fast call then

Reply:
Why not? We allow for v8::Local<v8::Value> in fastcalls, so it should work? Or am I missing something?